File: api/src/main/java/com/alten/remotesync/application/report/record/response/ReportDTO.java
package com.alten.remotesync.application.report.record.response;

import com.alten.remotesync.domain.report.enumeration.ReportStatus;
import com.alten.remotesync.domain.report.enumeration.ReportType;
import com.alten.remotesync.domain.user.model.User;

import java.time.LocalDateTime;
import java.util.UUID;

public record ReportDTO(
        UUID reportId,

        String title,

        String reason,

        ReportType type,

        ReportStatus status,

        String description,

        LocalDateTime createdAt,

        LocalDateTime updatedAt,

        User createdBy,

        User updatedBy



) {
}

File: app/src/app/dto/report.dto.ts
import { ReportType } from './report-type.enum';
import { ReportStatus } from './report-status.enum';
import { User } from './user.dto';

export interface ReportDTO {
  reportId: string;
  title: string;
  reason: string;
  type: ReportType;
  status: ReportStatus;
  description: string;
  createdAt: string;
  updatedAt: string;
  createdBy: User;
  updatedBy: User;
}
